critiquebrainz/frontend/external/bookbrainz_db/identifiers.py

```diff
@@ -0,0 +1,123 @@
+from typing import List
+from brainzutils import cache
+import sqlalchemy
+import critiquebrainz.frontend.external.bookbrainz_db as db
+from critiquebrainz.frontend.external.bookbrainz_db import DEFAULT_CACHE_EXPIRATION
+
+def fetch_identifiers(identifier_set_id: int) -> List:
+    """
+    Fetch identifiers from the database.
+    Args:
+        identifier_set_id (int): Identifier set ID.
+    Returns:
+        List of identifiers.
+    """
+    if not identifier_set_id:
+        return None
+    
+    key = cache.gen_key('identifier', identifier_set_id)
+    identifiers = cache.get(key)
+    if not identifiers:
+        with db.bb_engine.connect() as connection:
+            result = connection.execute(sqlalchemy.text("""
+                SELECT
+                    iden.type_id as type_id,
+                    idtype.label as label,
+                    iden.value as value
+                FROM bookbrainz.identifier_set__identifier idens
+                LEFT JOIN bookbrainz.identifier iden on idens.identifier_id = iden.id
+                LEFT JOIN bookbrainz.identifier_type idtype on iden.type_id = idtype.id
+                WHERE idens.set_id = :identifier_set_id;
+                """), ({'identifier_set_id': identifier_set_id}))
+            identifiers = result.fetchall()
+            identifiers = [dict(identifier) for identifier in identifiers]
+            identifiers = process_identifiers(identifiers)
+            cache.set(key, identifiers, DEFAULT_CACHE_EXPIRATION)
+
+    if not identifiers:
+        return None
+    return identifiers
+
+
+def process_identifiers(identifiers: List) -> List:
+    """Process identifiers and include urls."""
+    external_urls = []
+
+    url_map = {
+        1: "https://musicbrainz.org/release/",
```

Comment:
This map is OK for now, as it copies [data that is already in the BB website](https://github.com/metabrainz/bookbrainz-site/blob/master/src/client/components/pages/entities/identifiers-links.js). However, I don't like having data that can potentially change in two different locations (BB and CB). In the case that a new identifier is added to BB, we also need to remember to add it here, and I guarantee that we will forget to do so.

In BB, we already have the `bookbrainz.identifier_type` table, which includes these identifiers _and already has a column for a recognition regex_. I'd like to open a discussion with @MonkeyDo to add a new column to this table to also have the "construct url" text, so that we can share this data between BB and CB by using the database.

Can you please open a ticket in Jira for the CB project for an enhancement to read these identifiers from the database?